src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,99 @@
 /* eslint-disable */
 import React from 'react';
+import classNames from 'classnames';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Todo } from '../../types/Todo';
+import { actions as currentTodoActions} from '../../features/currentTodo';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const filter = useAppSelector(state => state.filter);
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  const filteredTodos = todos.filter(todo => {
+    if (
+      filter.query &&
+      !todo.title.toLowerCase().includes(filter.query.toLowerCase())
+    ) {
+      return false;
+    }
+
+    switch (filter.status) {
+      case 'active':
+        return !todo.completed;
+
+      case 'completed':
+        return todo.completed;
+
+      default:
+        return true;
+    }
```

Comment:
move filter inside switch case to avoid extra iteration when filter.status is `all`